openid/src/main/java/com/inrupt/client/openid/OpenIdSession.java

```diff
@@ -239,19 +239,25 @@ public void reset() {
     @Override
     public CompletionStage<Optional<Credential>> authenticate(final Authenticator auth,
             final Request request, final Set<String> algorithms) {
-        final Optional<Credential> cred = getCredential(ID_TOKEN, null);
-        if (cred.isPresent() && request != null) {
-            LOGGER.debug("Setting cache entry for request: {}", request.uri());
-            requestCache.put(cacheKey(request.uri()), Boolean.TRUE);
-        }
-        return CompletableFuture.completedFuture(cred);
+        return auth.authenticate(this, request, algorithms)
+                .thenApply(credential -> {
+                    if (credential != null) {
+                        requestCache.put(request.uri(), credential);
+                    }
+                    return Optional.ofNullable(credential);
+                });
     }
 
     /* deprecated */
     @Override
     public CompletionStage<Optional<Credential>> authenticate(final Request request,
             final Set<String> algorithms) {
-        return authenticate(null, request, algorithms);
+        final Optional<Credential> cred = getCredential(ID_TOKEN, null);
+        if (cred.isPresent() && request != null) {
+            LOGGER.debug("Setting cache entry for request: {}", request.uri());
+            requestCache.put(request.uri(), cred.get());
```

Comment:
```suggestion
            requestCache.put(cacheKey(request.uri()), cred.get());
```

Reply:
Fixed in cddd3e9bde469a93644c38a75b41643632055b6f